assets/js/tablegenerator.js

```diff
@@ -8,9 +10,13 @@ window.onload(function() {
     initialize()
 })
 
-async function getClassList() {
+
+
+async function fetchClassList() {
+    // Create array of classes
     var classes = []
 
+    // Fetch backend
     try {
```

Comment:
Not sure if the fetch request should be in a try, catch loop unless you have a specific reason for it to be. Not 100% but it looks like it's just if you aren't signed in which is fine, but maybe there's a way to check that before the fetch request goes through so we don't miss other possible errors in the try, catch.